include/scalar_node.hh

```diff
@@ -293,82 +425,117 @@ struct _mul_ : public ScalarNode {
 struct _div_ : public ScalarNode {
 	static const char op_code = 5;
 	static const char n_eval_args = 3;
-	inline static void eval(double &res, double a, double b) {
+	template <typename VecType>
+	inline static void eval(VecType &res, VecType a, VecType b) {
 		res = a / b;
 	}
 };
 
 struct _mod_ : public ScalarNode {
 	static const char op_code = 6;
 	static const char n_eval_args = 3;
-	inline static void eval(double &res, double a, double b) {
-		// TODO: vectorize
-		res = std::fmod(a, b);
-	}
+	template <typename VecType>
+	inline static void eval(VecType &res, VecType a, VecType b);
 };
+template<typename VecType>
+inline void _mod_::eval(VecType &res, VecType a, VecType b) {
+	res = a - b * truncate(a / b);
+}
+template<>
+inline void _mod_::eval<double>(double &res, double a, double b) {
+	res = std::fmod(a, b);
+}
 
 struct _eq_ : public ScalarNode {
 	static const char op_code = 7;
 	static const char n_eval_args = 3;
-	inline static void eval(double &res, double a, double b) {
-		// TODO: vectorize
-		res = double_bool(a == b);
-	}
+	template <typename VecType>
+	inline static void eval(VecType &res, VecType a, VecType b);
 };
+template<typename VecType>
```

Comment:
Specializovat `as_double` aby se to nemuselo specializovat zde a v dalších porovnávacích operací.